### PR TITLE
feat: SReLU and ActivityRegularization support

### PR DESCRIPTION
**Pull request by i8run**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this patch)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If it is possible, please attach a screenshot; otherwise, remove this)

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

**Comment by yangw1234** [comment]
LGTM

**Comment by zhichao-li** [comment]
@wzhongyuan  How can we make sure the new added layer match the Model serialization logic?

**Comment by zhichao-li** [comment]
@i8run `ActivityRegularization` should be exposed as a Layer, then we can add it like the other Layer into the Model instead of just use it as Layer parameter.
i.e
```
    layer = ActivityRegularization(l1=0.01, l2=0.01)

    # test in functional API
    x = Input(shape=(3,))
    z = Dense(2)(x)
    y = layer(z)
    model = Model(input=x, output=y)
```

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/SReLUSpec.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    srelu.forward(input)
+    println(srelu.output)
+  }
+
+  "SReLU serialize" should "work correctly" in {
+    val srelu = SReLU[Float]()
+    val input = Tensor[Float](5, 2, 3, 4).randn()
+    val res1 = srelu.forward(input)
+
+    ModulePersister.saveToFile[Float]("/tmp/srelu.bigdl", srelu, true)
```
Comment (wzhongyuan) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/SReLUSpec.scala]: Can we move it into 'ModuleSerialierSpec' with an separate unit test ?

Comment (zhichao-li) [reply spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/SReLUSpec.scala]: What I'm thinking is if we can make it be a common method, then the new added layer can be tested automatically for serialization which would make life more easier.
It's fine if move it to other spec.

Comment (wzhongyuan) [reply spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/SReLUSpec.scala]: @zhichao-li  That's the next plan, I will take care of it

**Comment by wzhongyuan** [comment]
LGTM

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SReLU.scala [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+  override def updateGradInput(input: Tensor[T], gradOutput: Tensor[T]): Tensor[T] = {
+    gradInput.resizeAs(input)
+
+    var batch = 0
+    while (batch < gradInput.size(1)) {
+      val sliceInput = input.select(1, batch + 1)
+      val sliceGradInput = gradInput.select(1, batch + 1)
+      val sliceGradOutput = gradOutput.select(1, batch + 1)
+
+      val xArr = sliceInput.storage().array()
```
Comment (yangw1234) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SReLU.scala]: Is it possible that the input tensor is not contiguous?

**Comment by yangw1234** [comment]
LGTM

**Pull request merged**